packages/hardhat-core/src/internal/core/typescript-support.ts

```diff
@@ -1,9 +1,28 @@
-import chalk from "chalk";
+import { HardhatConfig } from "../../types";
 
-import { isRunningTests } from "./execution-mode";
+import { resolveConfigPath } from "./config/config-loading";
+import { HardhatError } from "./errors";
+import { ERRORS } from "./errors-list";
+import { isRunningHardhatCoreTests } from "./execution-mode";
 
 let cachedIsTypescriptSupported: boolean | undefined;
 
+/**
+ * Returns true if Hardhat will run in using typescript mode.
+ * @param configPath The config path if provider by the user.
+ */
+export function willRunWithTypescript(configPath?: string): boolean {
+  const config = resolveConfigPath(configPath);
+  return isTypescriptFile(config);
+}
+
+/**
+ * Returns true if an Hardhat is already running with typescript.
+ */
+export function isRunningWithTypescript(config: HardhatConfig): boolean {
```

Comment:
This function seems to only be used in the tests

Reply:
It's used in the `test` task. I'm not sure if it should receive the config tbh. Maybe the HRE is better?

Reply:
Oh, my bad, I read "test" in the list of files that referenced it and didn't notice it was the task :sweat_smile: 

> I'm not sure if it should receive the config tbh. Maybe the HRE is better?

I think the config is fine. In general, passing the HRE feels like _too much_ and I think we should avoid it (I don't have a strong argument, it just feels wrong to pass an object that can do a lot of things).